### DISCUSSION TITLE
Update instead of just create

### DISCUSSION THREAD

**Discussion by elfeffe** [konnco/filament-import#48] [Ideas]
We can pass some ID, to update the model with the new data, instead of just create.
.

**Comment by frankyso** [comment]
currently its not supported for that, but think it can be an idea so everybody can override the creating functions.

**Comment by frankyso** [comment]
> We can pass some ID, to update the model with the new data, instead of just create. .

https://github.com/konnco/filament-import/releases/tag/1.4.0
you can try this version, i have added `handleRecordCreation` function. please give a feedback when you find a bug.

**Comment by elfeffe** [comment]
I get:

`Konnco\FilamentImport\Import::doMutateAfterCreate(): Argument #1 ($model) must be of type Illuminate\Database\Eloquent\Model, null given, called in /media/psf/www/filament-commerce/packages/elfeffe/filament-import/src/Import.php on line 203`

It's because of the empty rows that I have mentioned here https://github.com/konnco/filament-import/issues/50

It can be fixed using filter() here:

```
 public function getSpreadsheetData(): Collection
    {
        return $this->toCollection(new UploadedFile(Storage::disk($this->disk)->path($this->spreadsheet), $this->spreadsheet))
            ->filter()
            ->first()
            ->skip((int) $this->shouldSkipHeader);
    }
```

But when updating models, an ID can be wrong int he excel and the model could not exist.
The rest is working using this closure:
```

->handleRecordCreation(function ($prepareInsert) {
                    $model = Product::find($prepareInsert['id']);

                    if($model)
                    {
                        $model->fill($prepareInsert);
                        $model->save();

                    } else {
                        Notification::make()
                            ->danger()
                            ->title('Model with ID ' . $prepareInsert['id'] . ' not found.')
                            ->persistent()
                            ->send();
                    }

                    return $model;
                })
```

**Reply by frankyso** [reply]
I still don't really understand, so the file, you are trying to import has an empty row? or is it doesn't detect the rows data?

**Comment by elfeffe** [comment]
I don’t know. For some reason the excel file that I have to import has a
lot of empty rows at the end.
I have downloaded it from Google Spreadsheet, but I’m sure I’m not the only
one.

I have 200 rows with data, but the excel file has 999, probably a standard
in google docs.

**Reply by frankyso** [reply]
ahh okay understood, will check on that

**Comment by sergioraldana** [comment]
I have the same problem, I tried ->handleBlankRows(true) but get:

Method Konnco\FilamentImport\Actions\ImportAction::handleBlankRows does not exist.

My problem is that $model is null  by design because the $data[id] does not exists.

**Reply by frankyso** [reply]
try to update the package

**Reply by sergioraldana** [reply]
thanks, now I get: Undefined array key "profesion"

**Reply by tonypartridger** [reply]
You need to map your data. check the array key exists before calling profesion like:

$row['profesion'] = isset($row['profesion']) ? $row['profesion'] :  null;

**Comment by ocarlosapenas** [comment]
Guys, I've made a PR creating a configuration for the package that gives the option to update data, not only insert. I saw that I could achieve this using the handleRecordCreation closure, but I don't want to replicate this on all pages where I'm using the import feature. So please take a look at PR [#110 ](https://github.com/konnco/filament-import/pull/111)
